skorch/tests/test_callbacks.py

```diff
@@ -477,32 +649,36 @@ def test_args_passed_to_tabulate(self, history):
             assert tab.call_args_list[0][1]['floatfmt'] == '.9f'
 
     def test_with_additional_key(self, history, print_log_cls):
-        keys_ignored = ['batches']  # 'text' and 'dur' no longer ignored
+        keys_ignored = ['batches']  # 'dur' no longer ignored
         print_log = print_log_cls(
             sink=Mock(), keys_ignored=keys_ignored).initialize()
         # does not raise
         print_log.on_epoch_end(Mock(history=history))
 
         header = print_log.sink.call_args_list[0][0][0]
         columns = header.split()
-        expected = ['epoch', 'text', 'train_loss', 'valid_loss', 'dur']
+        expected = ['epoch', 'nmse', 'train_loss', 'valid_loss', 'dur']
         assert columns == expected
 
     def test_keys_ignored_as_str(self, print_log_cls):
         print_log = print_log_cls(keys_ignored='a-key')
         assert print_log.keys_ignored == ['a-key']
 
-    def test_no_valid(self, train_loss, valid_loss, print_log, ansi):
-        get_history(train_loss, valid_loss, print_log, with_valid=False)
+    def test_witout_valid_data(
+            self, net_cls, module_cls, mse_scoring, print_log, data):
+        net = net_cls(
+            module_cls, batch_size=1, train_split=None,
+            callbacks=[mse_scoring], max_epochs=2)
+        net.initialize()
+        # replace default PrintLog with test PrintLog
+        net.callbacks_[-1] = ('print_log', print_log)
+        net.partial_fit(*data)
+
         sink = print_log.sink
         row = sink.call_args_list[2][0][0]
         items = row.split()
 
-        assert len(items) == 2  # no valid
-        # epoch
-        assert items[0] == '1'
-        # color 1 used for item 1
-        assert items[1] == list(ansi)[1].value + '0.2500' + ansi.ENDC.value
```

Comment:
The code is gone now it for the future I would suggest not to test side effects like this but to refactor the coloring function and test it separately in a parametrised test